File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    let dp = [1, 2];
    
    if (n < 2) return dp[n - 1];
    
    for(let i = 3; i <= n; i++){
        dp.push((dp.at(-1) + dp.at(-2)) % 1234567)
    }
    
    return dp.at(-1)
    
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
import sys
sys.setrecursionlimit(10 ** 9)

dp = [0, 1, 2]

def recursive(n):
    for i in range(n + 1):
        if i < len(dp):
            continue
        dp.append((dp[-1] + dp[-2]) % 1234567)
    return dp[n]
        

def solution(n):
    return recursive(n)